script/publishing-api-sync-checks/detailed_guide_sync_check.rb

```diff
@@ -0,0 +1,78 @@
+detailed_guides = DetailedGuide.latest_published_edition
+check = DataHygiene::PublishingApiSyncCheck.new(detailed_guides)
+
+check.add_expectation("format") do |content_store_payload, _|
+  content_store_payload["format"] == "detailed_guide"
+end
+
+check.add_expectation("base_path") do |content_store_payload, record|
+  content_store_payload["base_path"] == record.search_link
+end
+
+check.add_expectation("title") do |content_store_payload, record|
+  content_store_payload["title"].strip == record.title.strip
+end
+
+check.add_expectation("related_guides") do |content_store_payload, record|
+  payload_ids = content_store_payload["links"]["related_guides"]
```

Comment:
I don't think that's golfish, `to_set` is what I would have used had I known it existed!

I'm going to do something like:

```
payload_ids = content_store_payload["links"]["related_guides"].map { |rg| rg["content_id"] }.to_set
payload_ids == record.related_detailed_guide_content_ids.to_set
```

Which is terse and also keeps line length down.